src/naive/period.rs

```diff
@@ -0,0 +1,279 @@
+use naive::NaiveDateTime;
+use oldtime::Duration;
+
+use std::cmp::{min, max};
+
+/// A period of time between two ISO 8601 dates/times ([NaiveDateTime](NaiveDateTime)s).
+/// This is similar to a [`Duration`](Duration), except that it has a fixed start
+/// date/time (and thus a fixed end date/time), allowing clients to determine if specific
+/// segments of time intersect.
+///
+/// # Notes
+/// This period is not reliant on time zones. For the time being, offsets aren't considered at all.
+/// As such, if you need to interset two time periods that are of differing time zones, you're out
+/// of luck (patches accepted, though).
+#[derive(PartialEq, Eq, PartialOrd, Ord, Copy, Clone, Debug)]
+pub struct NaivePeriod {
```

Comment:
Can we accomplish a bit more future compatibility and smaller api surface with a `Period<T: Datelike + Timelike>`? I'd rather reduce the number of places that chrono distinguishes between Naive and tz-aware types. Putting this behind an unstable flag will allow a punt on this for awhile.